src/utils/file.ts

```diff
@@ -0,0 +1,22 @@
+import * as fs from 'fs'
+
+export default class File {
```

Comment:
- [ ] I don't think that a `File` abstraction is needed. You can always using [`path.format`](https://nodejs.org/api/path.html#path_path_format_pathobject) plus a promisified version of `writeFile`:

```javascript
import { format } from 'path';
import { writeFile } from 'fs';
import { promisify } from 'util';

async () => {
    const writeF = promisify( writeFile );
    await writeF( 'my-file.json', '{ "foo": "bar" }' );
    await writeF( format( { name: 'my-file', ext: '.json' } ), '{ "bar": "foo" }' );
}
```